front/src/applications/operationalStudies/components/MacroEditor/types.ts

```diff
@@ -65,6 +65,30 @@ export type TimeLock = {
   timeFormatter: null;
 };
 
+export type TrainrunSectionOperation = {
```

Comment:
I think we can just use the existing `TrainrunSection` type here instead of re-defining a new one?

Reply:
We thought `TrainrunSection` was used for typed TrainrunSectionDTO.
We can therefore add keys to `TrainrunSection` ?

Reply:
Eh, indeed, there are differences between [the DTO type](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/blob/b83f0ec2af0c4658632d03ac4497c6a38807cb75/src/app/data-structures/business.data.structures.ts#L113) and [the model type](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/blob/b83f0ec2af0c4658632d03ac4497c6a38807cb75/src/app/models/trainrunsection.model.ts#L19)...

Which additional fields do we need that are missing from the DTO? If we need these additional fields, I agree it's better to keep the types separate to not add extraneous fields which won't be taken into account by NGE to the DTO.

Reply:
We need `targetNode`, `sourceNode` and `trainrun`

Reply:
Since the `operations` need more detail, we could stick to the model type, and create a new DTO type for `netzgrafikDto`

Reply:
Hm. Howe about we do this for now?

```ts
export type OperationTrainrunSection = TrainrunSection & {
  // extra props we need here
};
```

Nit: IMHO `OperationTrainrunSection` reads a _tad_ better than `TrainrunSectionOperation`: the latter makes it sound like the whole thing is an operation.

Reply:
Done